File: POOBanca/C++/Banca.cpp
#include "Banca.h"

// Método comum sobreposto
void Banca::apresentacao() {
    cout << "Os produtos da banca são:" << endl;

    if(this->prod1 != NULL) {
        this->prod1->apresentacao();
    }

    if(this->prod2 != NULL) {
        this->prod2->apresentacao();
    }

    if(this->prod3 != NULL) {
        this->prod3->apresentacao();
    }

    if(this->prod4 != NULL) {
        this->prod4->apresentacao();
    }

    if(this->prod5 != NULL) {
        this->prod5->apresentacao();
    }
}

// Construtores e Destrutor
Banca::Banca(Produto *prod1) {
    this->prod1 = prod1;
    this->prod2 = NULL;
    this->prod3 = NULL;
    this->prod4 = NULL;
    this->prod5 = NULL;
}

Banca::Banca(Produto *prod1, Produto *prod2) {
    this->prod1 = prod1;
    this->prod2 = prod2;
    this->prod3 = NULL;
    this->prod4 = NULL;
    this->prod5 = NULL;
}

Banca::Banca(Produto *prod1, Produto *prod2, Produto *prod3) {
    this->prod1 = prod1;
    this->prod2 = prod2;
    this->prod3 = prod3;
    this->prod4 = NULL;
    this->prod5 = NULL;
}

Banca::Banca(Produto *prod1, Produto *prod2, Produto *prod3, Produto *prod4) {
    this->prod1 = prod1;
    this->prod2 = prod2;
    this->prod3 = prod3;
    this->prod4 = prod4;
    this->prod5 = NULL;
}

Banca::Banca(Produto *prod1, Produto *prod2, Produto *prod3, Produto *prod4, Produto *prod5) {
    this->prod1 = prod1;
    this->prod2 = prod2;
    this->prod3 = prod3;
    this->prod4 = prod4;
    this->prod5 = prod5;
}

Banca::~Banca() {
    delete this->prod1;
    delete this->prod2;
    delete this->prod3;
    delete this->prod4;
    delete this->prod5;

    cout << "Objeto Banca Destruído com Sucesso!" << endl;
}

// Getters
const Produto *Banca::getProd1() {
    return this->prod1;
}

const Produto *Banca::getProd2() {
    return this->prod2;
}

const Produto *Banca::getProd3() {
    return this->prod3;
}

const Produto *Banca::getProd4() {
    return this->prod4;
}

const Produto *Banca::getProd5() {
    return this->prod5;
}

// Setters
void Banca::setProd1(Produto *prod1) {
    this->prod1 = prod1;
}

void Banca::setProd2(Produto *prod2) {
    this->prod2 = prod2;
}

void Banca::setProd3(Produto *prod3) {
    this->prod3 = prod3;
}

void Banca::setProd4(Produto *prod4) {
    this->prod4 = prod4;
}

void Banca::setProd5(Produto *prod5) {
    this->prod5 = prod5;
}

File: POOBanca/Python/Banca.py
from Produto import Produto
from Ponto import Ponto

class Banca(Ponto):
    __prod1 = None
    __prod2 = None
    __prod3 = None
    __prod4 = None
    __prod5 = None

    def apresentacao(self):
        print("Os produtos da banca são:")

        if self.__prod1 is not None:
            self.__prod1.apresentacao()

        if self.__prod2 is not None:
            self.__prod2.apresentacao()

        if self.__prod3 is not None:
            self.__prod3.apresentacao()

        if self.__prod4 is not None:
            self.__prod4.apresentacao()

        if self.__prod5 is not None:
            self.__prod5.apresentacao()

    def __init__(self, prod1, prod2 = None, prod3 = None, prod4 = None, prod5 = None):
        self.__prod1 = prod1
        self.__prod2 = prod2
        self.__prod3 = prod3
        self.__prod4 = prod4
        self.__prod5 = prod5

    def __del__(self):
        del self.__prod1
        del self.__prod2
        del self.__prod3
        del self.__prod4
        del self.__prod5

        print("Objeto Banca Destruído com Sucesso!")

    def getProd1(self):
        return self.__prod1

    def getProd2(self):
        return self.__prod2

    def getProd3(self):
        return self.__prod3

    def getProd4(self):
        return self.__prod4

    def getProd5(self):
        return self.__prod5

    def setProd1(self, prod1):
        self.__prod1 = prod1

    def setProd2(self, prod2):
        self.__prod2 = prod2

    def setProd3(self, prod3):
        self.__prod3 = prod3

    def setProd4(self, prod4):
        self.__prod4 = prod4

    def setProd5(self, prod5):
        self.__prod5 = prod5